app/controllers/admin/schedule_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Admin
+  class ScheduleController < Admin::BaseController
+    # By authorizing 'conference' resource, we can ensure there will be no unauthorized access to
+    # the schedule of a conference, which should not be accessed in the first place
+    load_and_authorize_resource :schedule
+    load_and_authorize_resource :conference, find_by: :short_title
+    load_and_authorize_resource :program, through: :conference, singleton: true
+    load_resource :venue, through: :conference, singleton: true
+
+    skip_before_action :verify_authenticity_token, only: [:update]
+
+    def index
+      @schedules = @conference.program.schedules
+      @selected_schedule = @conference.program.selected_schedule
+    end
+
+    def create
+      new_schedule = @program.schedules.create
+      redirect_to action: 'show', id: new_schedule.id
+    end
+
+    def show
+      @schedule_id = params[:id].to_i
+      schedule = Schedule.find(@schedule_id)
+      @event_schedules = schedule.event_schedules
```

Comment:
You could cover this also with a before filter

``
 load_and_authorize_resource :event_schedules, through: :schedule 
``